### ISSUE TITLE
New Rule: No invalid JSON args

### ISSUE THREAD

**Issue by brunocrosier** [eslint/eslint#16371]
### Rule details

Ensure the arguments passed to `JSON.parse()` and `JSON.stringify()` are valid

### Related ECMAScript feature

JSON

### What type of rule is this?

Warns about a potential problem

### Example code

```js
JSON.parse("")
JSON.parse("'Lorem ipsum'")
JSON.parse("[1, 2, 3, 4, ]");
JSON.parse('{"foo" : 1, }');
JSON.parse("{'foo': 1}");

JSON.stringify(2n);
JSON.stringify({
  deeply: {
    nested: {
      value: 2n
    }
  }
});
```

### Why should this rule be in the core instead of a plugin?

If this were a plugin, there would be nothing to configure. The arguments and return values of `JSON.parse()` and `JSON.stringify()` are predictable.

I cannot envisage any scenario in which a developer would benefit from not having this rule enabled.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by mdjermanovic** [comment]
Hi @brunocrosier, thanks for the issue!

I think that calling JSON methods with arguments that can be statically determined, such as `JSON.parse(<string literal>)`, is a rare use case for these methods, and that therefore this rule doesn't qualify for inclusion in the core. If this is a common use case in your codebase, you can always make a custom rule.
